src/functionalTest/java/uk/gov/hmcts/reform/iacaseapi/util/SystemDocumentManagementUploader.java

```diff
@@ -0,0 +1,85 @@
+package uk.gov.hmcts.reform.iacaseapi.util;
+
+import com.google.common.io.ByteStreams;
+import java.io.IOException;
+import java.util.Collections;
+import org.springframework.core.io.Resource;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+import uk.gov.hmcts.reform.document.DocumentUploadClientApi;
+import uk.gov.hmcts.reform.document.domain.UploadResponse;
+import uk.gov.hmcts.reform.document.utils.InMemoryMultipartFile;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.field.Document;
+
+@Service
+public class SystemDocumentManagementUploader {
+
+    private final DocumentUploadClientApi documentUploadClientApi;
+    private final AuthorizationHeadersProvider authorizationHeadersProvider;
+
+    public SystemDocumentManagementUploader(
+        DocumentUploadClientApi documentUploadClientApi,
+        AuthorizationHeadersProvider authorizationHeadersProvider
+    ) {
+        this.documentUploadClientApi = documentUploadClientApi;
+        this.authorizationHeadersProvider = authorizationHeadersProvider;
+    }
+
+    public Document upload(
+        Resource resource,
+        String contentType
+    ) {
+        final String serviceAuthorizationToken =
+            authorizationHeadersProvider
+                .getLegalRepresentativeAuthorization()
+                .getValue("ServiceAuthorization");
+
+        final String accessToken =
+            authorizationHeadersProvider
+                .getLegalRepresentativeAuthorization()
+                .getValue("Authorization");
+
+        final String userId = "1";
+
+        try {
+
+            MultipartFile file = new InMemoryMultipartFile(
+                resource.getFilename(),
+                resource.getFilename(),
+                contentType,
+                ByteStreams.toByteArray(resource.getInputStream())
+            );
+
+            UploadResponse uploadResponse =
+                documentUploadClientApi
+                    .upload(
+                        accessToken,
+                        serviceAuthorizationToken,
+                        userId,
+                        Collections.singletonList(file)
+                    );
+
+            uk.gov.hmcts.reform.document.domain.Document uploadedDocument =
```

Comment:
Is there another Class name that would make sense to avoid fully qualified class name?

Reply:
This is from a library we use from `document-management-client` which clashed with our own `Document` object name.  I think since it's just in one place I think it's acceptable.  Later we can consider refactoring our internal `Document` to be named something like `CaseDocument` or similar.